rero_ils/modules/documents/tasks.py

```diff
@@ -36,6 +36,43 @@ def reindex_document(pid):
     Document.get_record_by_pid(pid).reindex()
 
 
+@shared_task(ignore_result=True)
+def delete_orphan_harvested(delete=False, verbose=False):
+    """Delete orphan harvested documents.
+
+    :param delete: if True delete from DB and ES.
+    :param verbose: Verbose print.
+    :returns: count of deleted documents.
+    """
+    query = (
+        DocumentsSearch()
+        .filter("term", harvested=True)
+        .exclude("exists", field="holdings")
+    )
+    pids = [hit.pid for hit in query.source("pid").scan()]
+    count = len(pids)
+
+    if verbose:
+        click.secho(f"Orphan harvested documents count: {count}", fg="yellow")
+    for pid in pids:
+        if doc := Document.get_record_by_pid(pid):
+            if verbose:
+                click.secho(f"Deleting orphan harvested: {pid}", fg="yellow")
+            if delete:
+                try:
+                    if doc.reasons_not_to_delete() == {"others": {"harvested": True}}:
+                        doc.delete(dbcommit=True, delindex=True, force=True)
```

Comment:
Not sure `force=True` will do the trick. Force will delete the persistent identifier in DB. I think yo have to set `harvested` to `False` before deleting.